tests/test_spam_admin.py

```diff
@@ -0,0 +1,440 @@
+from __future__ import absolute_import
+from nose.tools import *  # noqa PEP8 asserts
+
+
+from framework.auth import Auth
+from website.project.model import Comment
+
+from website.settings import SPAM_ASSASSIN
+from tests.base import (
+    OsfTestCase,
+)
+from tests.factories import (
+    ProjectFactory, AuthUserFactory
+)
+from framework.exceptions import HTTPError
+from website.spam_admin.spam_admin_settings import SPAM_ASSASSIN_URL,SPAM_ASSASSIN_TEACHING_URL
+import httpretty
+from website.spam_admin.utils import train_spam, _project_is_spam
+from website.project.model import Node
+from website.project.views.node import project_before_set_public
+import json
+
+
+class TestCommentSpamAdmin(OsfTestCase):
+
+
+    def setUp(self):
+        super(TestCommentSpamAdmin, self).setUp()
+        self.project = ProjectFactory(is_public=True)
+        self.project.save()
+
+        #spam assassin
+        self.GTUBE = "SPAM"
+        self.spam_assassin_active = SPAM_ASSASSIN
+        self.spam_admin = AuthUserFactory(spam_admin=True)
+
+    def _configure_project(self, project, comment_level):
+
+        project.comment_level = comment_level
+        project.save()
+
+
+    def _add_comment(self, project, content=None, **kwargs):
+        def request_callback(request, uri, headers):
+            if self.GTUBE in request.body:
+                return (200, headers, json.dumps({"decision":"SPAM"}))
+            return (200, headers, json.dumps({"decision":"HAM"}))
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_URL,
+            body=request_callback
+        )
+
+        content = content if content is not None else 'hammer to fall'
+        url = project.api_url + 'comment/'
+        ret = self.app.post_json(
+            url,
+            {
+                'content': content,
+                'isPublic': 'public',
+                'page': 'node',
+                'target': project._id
+            },
+            **kwargs
+        )
+
+        return Comment.load(ret.json['comment']['id'])
+
+
+    ########################## TEST MODEL / UTIL FUNCTIONS  #####################################################
+
+
+    @httpretty.activate
+    def test_comment_added_is_spam(self):
+        comment = self._add_comment(
+            self.project, content = self.GTUBE, auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+
+        assert_equal(len(self.project.commented), 1)
+
+        if self.spam_assassin_active:
```

Comment:
I am not sure how to do this. Spam Assassin could be active or inactive depending on the settings as defined in website.settings. These settings affect how many of the functions will work. I cannot change what the settings are from inside these tests. Therefore, I decided to just use an if-then to check what occurs in either case.